### PR TITLE
refactor: review fixes and improvements

### PR DESCRIPTION
**Pull request by evilrobot-01**
Various fixes and improvements from reviewing #371

### CODE REVIEW

File: crates/pop-cli/src/commands/up/contract.rs [hunk 1]
```diff
@@ -194,14 +194,18 @@ impl UpContractCommand {
 				spinner.start("Uploading contract...");
 
 				if self.upload_only {
-					let result = upload_contract_signed(self.url.as_str(), payload).await;
```
Comment (evilrobot-01) [inline crates/pop-cli/src/commands/up/contract.rs:197]: Previous versions were less code, but resulted in the need for an expect/unwrap which is eliminated by using a match.

File: crates/pop-cli/src/commands/up/contract.rs [hunk 2]
```diff
@@ -344,7 +352,7 @@ impl UpContractCommand {
 
 	// get the call data and contract code hash
 	async fn get_contract_data(&self) -> anyhow::Result<(Vec<u8>, [u8; 32])> {
-		let contract_code = get_contract_code(self.path.as_ref()).await?;
```
Comment (evilrobot-01) [inline crates/pop-cli/src/commands/up/contract.rs:347]: Async unnecessary

File: crates/pop-cli/src/commands/up/contract.rs [hunk 3]
```diff
@@ -480,7 +488,6 @@ mod tests {
 	#[tokio::test]
 	async fn get_upload_and_instantiate_call_data_works() -> anyhow::Result<()> {
 		let (contracts_node_process, port, temp_dir) = start_test_environment().await?;
-		let localhost_url = format!("ws://127.0.0.1:{}", port);
```
Comment (evilrobot-01) [inline crates/pop-cli/src/commands/up/contract.rs:483]: Not used

File: crates/pop-cli/src/common/wallet.rs [hunk 4]
```diff
@@ -22,13 +22,12 @@ pub async fn request_signature(call_data: Vec<u8>, rpc: String) -> anyhow::Resul
 	let transaction_data = TransactionData::new(rpc, call_data);
 	// Starts server with port 9090.
 	let mut wallet = WalletIntegrationManager::new(ui, transaction_data, Some(9090));
-	let url = wallet.server_url.clone();
```
Comment (evilrobot-01) [inline crates/pop-cli/src/common/wallet.rs:25]: Removes a few unnecessary clones and simplifys the code.

**Review by AlexD10S — APPROVED** [review]
Looks good. Thanks for taking care of this!

**Pull request merged**